napari/components/viewer_model.py

```diff
@@ -557,6 +557,7 @@ def add_layer(self, layer: layers.Layer) -> layers.Layer:
         layer.events.data.connect(self._on_layers_change)
         layer.name = self.layers._coerce_name(layer.name, layer)
         layer.events.name.connect(self.layers._update_name)
+        layer.events.scale.connect(self._on_layers_change)
```

Comment:
Should we add for `translate`, `rotate`, `shear` and `affine` too? We might want to rethink this long term, but if we add for one, might as well add for all

Reply:
I meet this problem when update the scale. I'm not sure which are also need, so I'm waiting on such comment like Yours. 

I personally prefer to add some event which should be emitted on every change which should cause the call of function like `_on_layers_change`. It's maybe useful with also some synchronization etc. 

But looking at the suggestion to release 0.4.2 on zulip I prefer to do a fast bugfix, and go back to refactor later.

Reply:
yup agreed, we should refactor later, i think this whole `_on_layers_change` stuff should change a lot